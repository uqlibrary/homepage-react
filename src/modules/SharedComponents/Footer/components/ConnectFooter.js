import React from 'react';
import PropTypes from 'prop-types';

import { default as locale } from '../footer.locale.js';
import { default as menuLocale } from 'locale/menu';

import Button from '@material-ui/core/Button';
import Tooltip from '@material-ui/core/Tooltip';
import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
    connectFooter: {
        fontWeight: '300',
        lineHeight: '25px',
        margin: '0 auto 0 auto',
        maxWidth: '90%',
        [theme.breakpoints.up('lg')]: {
            maxWidth: '1200px',
        },
        padding: '20px 0',
        position: 'relative',
        '& a': {
            color: theme.palette.secondary.dark,
            textDecoration: 'none',
            '&:hover': {
                color: theme.palette.primary.main,
                textDecoration: 'underline',
            },
        },
    },
    navigation: {
        marginTop: '-4px',
        '& ul': {
            padding: 0,
            margin: 0,
            [theme.breakpoints.down('sm')]: {
                textAlign: 'center',
            },
        },
        '& li': {
            listStyle: 'none',
            margin: 0,
            padding: 0,
            [theme.breakpoints.down('sm')]: {
                display: 'inline-block',
            },
            [theme.breakpoints.up('md')]: {
                fontSize: 14,
                lineHeight: 1.7,
            },
        },
    },
    separator: {
        display: 'inline-block',
    },
    socialButtonClass: {
        width: 32,
        height: 32,
        minWidth: 32,
        padding: 0,
        backgroundColor: theme.palette.primary.main,
        color: theme.palette.white.main,
        '&:hover': {
            backgroundColor: theme.palette.primary.dark + '!important',
        },
    },
    internal: {
        bottom: '1rem',
        position: 'absolute',
        [theme.breakpoints.down('sm')]: {
            bottom: 'auto',
            marginTop: '1rem',
            position: 'relative',
            textAlign: 'center',
        },
    },
    giving: {
        '& div': {
            marginLeft: 'auto',
            [theme.breakpoints.down('sm')]: {
                margin: '5px auto',
                maxWidth: '300px',
            },
        },
        textAlign: 'right',
    },
    givingBlock: {
        maxWidth: 240,
        marginRight: 10,
    },
    givingButtonClass: {
        color: theme.palette.white.main + '!important',
        backgroundColor: theme.palette.accent.main,
        '&:hover': {
            backgroundColor: theme.palette.accent.dark,
        },
        padding: '1rem',
        textTransform: 'initial',
    },
    contacts: {
        marginTop: '-4px',
        maxWidth: 400,
        '& div': {
            '& div': {
                [theme.breakpoints.down('sm')]: {
                    margin: '0 auto',
                },
            },
        },
    },
});

export function ConnectFooter(props) {
    const { classes } = props;

    const loadLinkInSamePage = url => {
        window.location.assign(url);
    };

    const loadLinkToTarget = (url, target) => {
        window.open(url, target);
    };

    const separator = () => (
        <Hidden mdUp className={classes.separator}>
            &nbsp;|&nbsp;
        </Hidden>
    );

    return (
        <Grid
            className={classes.connectFooter}
            container
            data-testid="connect-footer"
            alignItems="flex-start"
            justify="center"
        >
            <Grid item xs={12} md={4} className={classes.navigation}>
                <ul>
                    <li>
                        <a data-testid="footermenu-homepage" href={menuLocale.menuhome.linkTo}>
                            {menuLocale.menuhome.primaryText}
                        </a>
                        {separator()}
                    </li>
                    {menuLocale.publicmenu.map((linkProperties, index) => (
                        <li key={`footerli-${index}`}>
                            <a data-testid={linkProperties.dataTestid} href={linkProperties.linkTo}>
                                {linkProperties.primaryText}
                            </a>
                            {index < menuLocale.publicmenu.length - 1 && separator()}
                        </li>
                    ))}
                </ul>
            </Grid>
            <Grid item xs={12} md={4} className={classes.contacts}>
                <Grid container>
                    <Grid item xs={'auto'}>
                        <Typography variant={'h6'} component={'h3'} style={{ marginTop: -4 }}>
                            {locale.connectFooter.buttonSocialHeader}
                        </Typography>
                    </Grid>
                </Grid>
                <Grid container spacing={1}>
                    {locale.connectFooter.buttonSocial.map((item, index) => (
                        <Grid item xs={'auto'} key={`buttonSocial-${index}`} id={`buttonSocial-${index}`}>
                            <Tooltip
                                id={`auth-button-${index}`}
                                title={`${item.linkMouseOver}`}
                                placement="bottom"
                                TransitionProps={{ timeout: 300 }}
                            >
                                <Button
                                    aria-label={item.linkMouseOver}
                                    classes={{
                                        root: classes.socialButtonClass,
                                    }}
                                    color="primary"
                                    variant="contained"
                                    data-testid={item.dataTestid}
                                    id={`socialbutton-${index}`}
                                    onClick={() => loadLinkToTarget(item.linkTo)}
                                >
                                    {item.icon}
                                </Button>
                            </Tooltip>
                        </Grid>
                    ))}
                </Grid>
                <Grid className={classes.internal}>
                    {locale.connectFooter.internalLinks.map((linkProperties, index) => {
                        return (
                            <span key={`internallabel-${index}`}>
                                <a data-testid={linkProperties.dataTestid} href={linkProperties.linkTo}>
                                    {linkProperties.linklabel}
                                </a>
                                {index < locale.connectFooter.internalLinks.length - 1 && <span>&nbsp;|&nbsp; </span>}
                            </span>
                        );
                    })}
                </Grid>
            </Grid>
            <Grid item xs={12} md={4} className={classes.giving}>
                <Grid container spacing={2}>
                    {locale.connectFooter.givingLinks.map((item, index) => {
                        return (
                            <Grid item xs={12} key={`givingLinks-${index}`} className={classes.givingBlock}>
                                <Button
                                    fullWidth
                                    children={item.label}
                                    className={classes.givingButtonClass}
                                    data-testid={item.dataTestid}
                                    key={`givingLinks-${index}`}
                                    onClick={() => loadLinkInSamePage(item.linkTo)}
                                    variant="contained"
                                />
                            </Grid>
                        );
                    })}
                </Grid>
            </Grid>
        </Grid>
    );
}

ConnectFooter.propTypes = {
    classes: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
};

ConnectFooter.defaultProps = {
    classes: {},
};

export default withStyles(styles, { withTheme: true })(ConnectFooter);
