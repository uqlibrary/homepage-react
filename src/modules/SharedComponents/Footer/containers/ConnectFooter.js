import { withRouter } from 'react-router-dom';

import ConnectFooter from '../components/ConnectFooter';

export default withRouter(ConnectFooter);
