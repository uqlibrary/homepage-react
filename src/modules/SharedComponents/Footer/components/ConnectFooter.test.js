Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
    })),
});

import { ConnectFooter } from './ConnectFooter';

function setup(testProps = {}) {
    const props = {
        ...testProps,
        classes: {},
        history: testProps.history || { push: jest.fn() },
    };
    return getElement(ConnectFooter, props);
}

describe('Component Megamenu', () => {
    it('should render', () => {
        const wrapper = setup();
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
