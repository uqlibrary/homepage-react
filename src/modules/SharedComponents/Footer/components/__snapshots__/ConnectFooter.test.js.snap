// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ConnectFooter should render 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="flex-start"
  container={true}
  data-testid="connect-footer"
  id="connect-footer-block"
  justify="center"
  style={
    Object {
      "display": "none",
    }
  }
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={4}
    xs={12}
  >
    <ul>
      <li>
        <a
          data-testid="footermenu-homepage"
          href="http://www.library.uq.edu.au"
        >
          Library home
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-0"
      >
        <a
          data-testid="connect-services-link"
          href="https://web.library.uq.edu.au/library-services"
        >
          Library services
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-1"
      >
        <a
          data-testid="connect-research-link"
          href="https://web.library.uq.edu.au/research-tools-techniques"
        >
          Research tools & techniques
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-2"
      >
        <a
          data-testid="connect-collections-link"
          href="https://web.library.uq.edu.au/collections"
        >
          Collections
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-3"
      >
        <a
          data-testid="connect-borrowing-link"
          href="https://web.library.uq.edu.au/borrowing-requesting"
        >
          Borrowing & requesting
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-4"
      >
        <a
          data-testid="connect-locations-link"
          href="https://web.library.uq.edu.au/locations-hours"
        >
          Locations & hours
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-5"
      >
        <a
          data-testid="connect-about-link"
          href="https://web.library.uq.edu.au/about-us"
        >
          About
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-6"
      >
        <a
          data-testid="connect-contact-link"
          href="https://web.library.uq.edu.au/contact-us"
        >
          Contact us
        </a>
      </li>
    </ul>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={4}
    xs={12}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs="auto"
      >
        <WithStyles(ForwardRef(Typography))
          component="h3"
          style={
            Object {
              "marginTop": -4,
            }
          }
          variant="h6"
        >
          Connect with us
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={1}
    >
      <WithStyles(ForwardRef(Grid))
        id="buttonSocial-0"
        item={true}
        key="buttonSocial-0"
        xs="auto"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionProps={
            Object {
              "timeout": 300,
            }
          }
          id="auth-button-0"
          placement="bottom"
          title="Library Blog"
        >
          <WithStyles(ForwardRef(Button))
            aria-label="Library Blog"
            classes={
              Object {
                "root": undefined,
              }
            }
            color="primary"
            data-testid="connect-blog-link"
            id="socialbutton-0"
            onClick={[Function]}
            variant="contained"
          >
            <CreateIcon />
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        id="buttonSocial-1"
        item={true}
        key="buttonSocial-1"
        xs="auto"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionProps={
            Object {
              "timeout": 300,
            }
          }
          id="auth-button-1"
          placement="bottom"
          title="Library on Twitter"
        >
          <WithStyles(ForwardRef(Button))
            aria-label="Library on Twitter"
            classes={
              Object {
                "root": undefined,
              }
            }
            color="primary"
            data-testid="connect-twitter-link"
            id="socialbutton-1"
            onClick={[Function]}
            variant="contained"
          >
            <TwitterIcon />
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        id="buttonSocial-2"
        item={true}
        key="buttonSocial-2"
        xs="auto"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionProps={
            Object {
              "timeout": 300,
            }
          }
          id="auth-button-2"
          placement="bottom"
          title="Library on Facebook"
        >
          <WithStyles(ForwardRef(Button))
            aria-label="Library on Facebook"
            classes={
              Object {
                "root": undefined,
              }
            }
            color="primary"
            data-testid="connect-facebook-link"
            id="socialbutton-2"
            onClick={[Function]}
            variant="contained"
          >
            <FacebookIcon />
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        id="buttonSocial-3"
        item={true}
        key="buttonSocial-3"
        xs="auto"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionProps={
            Object {
              "timeout": 300,
            }
          }
          id="auth-button-3"
          placement="bottom"
          title="Library on Instagram"
        >
          <WithStyles(ForwardRef(Button))
            aria-label="Library on Instagram"
            classes={
              Object {
                "root": undefined,
              }
            }
            color="primary"
            data-testid="connect-instagram-link"
            id="socialbutton-3"
            onClick={[Function]}
            variant="contained"
          >
            <InstagramIcon />
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        id="buttonSocial-4"
        item={true}
        key="buttonSocial-4"
        xs="auto"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionProps={
            Object {
              "timeout": 300,
            }
          }
          id="auth-button-4"
          placement="bottom"
          title="Library on YouTube"
        >
          <WithStyles(ForwardRef(Button))
            aria-label="Library on YouTube"
            classes={
              Object {
                "root": undefined,
              }
            }
            color="primary"
            data-testid="connect-youtube-link"
            id="socialbutton-4"
            onClick={[Function]}
            variant="contained"
          >
            <YouTubeIcon />
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))>
      <span
        key="internallabel-0"
      >
        <a
          data-testid="connect-feedback-link"
          href="https://support.my.uq.edu.au/app/library/feedback"
        >
          Library feedback
        </a>
        <span>
           |  
        </span>
      </span>
      <span
        key="internallabel-1"
      >
        <a
          data-testid="connect-participate-link"
          href="https://web.library.uq.edu.au/about-us/participate-customer-research"
        >
          Help us improve
        </a>
        <span>
           |  
        </span>
      </span>
      <span
        key="internallabel-2"
      >
        <a
          data-testid="connect-sitemap-link"
          href="https://web.library.uq.edu.au/sitemap"
        >
          Site Map
        </a>
      </span>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={4}
    xs={12}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="givingLinks-0"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          data-testid="connect-friends-link"
          fullWidth={true}
          key="givingLinks-0"
          onClick={[Function]}
          variant="contained"
        >
          Join Friends of the Library
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="givingLinks-1"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          data-testid="connect-give-link"
          fullWidth={true}
          key="givingLinks-1"
          onClick={[Function]}
          variant="contained"
        >
          Give to the Library
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
